packages/common/src/controllers/CheckoutController.ts

```diff
@@ -52,10 +53,14 @@ export default class CheckoutController {
       useCheckoutStore.setState({ subscriptionOffers });
     }
 
-    if (!useCheckoutStore.getState().switchSubscriptionOffers.length) {
-      const subscriptionSwitches = await this.getSubscriptionSwitches();
-      const switchSubscriptionOffers = subscriptionSwitches ? await this.getOffers({ offerIds: subscriptionSwitches }) : [];
-      useCheckoutStore.setState({ switchSubscriptionOffers });
+    try {
+      if (!useCheckoutStore.getState().switchSubscriptionOffers.length) {
+        const subscriptionSwitches = await this.getSubscriptionSwitches();
```

Comment:
Do you think it could make sense to return an empty array for `getSubscriptionSwitches` as this method is used as part of the init and we could expect to throw errors every time we open the page?

Reply:
Yes, you're right. I updated the `getSubscriptionSwitches` so it does not throw when it's not implemented.